src/psyclone/tests/gocean1p0_transformations_test.py

```diff
@@ -1184,3 +1184,196 @@ def test_module_inline_warning_no_change():
+
+    # Now swap the middle loops
+    swapped2, _ = swap.apply(swapped1.children[1])
+    psy.invokes.get('invoke_loop1').schedule = swapped2
+
+    assert str(psy.gen) == """  MODULE psy_test27_loop_swap
+    USE field_mod
+    USE kind_params_mod
+    IMPLICIT NONE
+    CONTAINS
+    SUBROUTINE invoke_loop1(cu_fld, p_fld, u_fld, unew_fld, uold_fld)
+      USE time_smooth_mod, ONLY: time_smooth_code
+      USE compute_cu_mod, ONLY: compute_cu_code
+      TYPE(r2d_field), intent(inout) :: cu_fld, p_fld, u_fld, unew_fld, uold_fld
+      INTEGER i
+      INTEGER j
+      INTEGER istop, jstop
+      !
+      ! Look-up loop bounds
+      istop = cu_fld%grid%simulation_domain%xstop
+      jstop = cu_fld%grid%simulation_domain%ystop
+      !
+      DO i=2,istop+1
+        DO j=2,jstop
+          CALL compute_cu_code(i, j, cu_fld%data, p_fld%data, u_fld%data)
+        END DO 
+      END DO 
+      DO i=2,istop+1
+        DO j=2,jstop
+          CALL compute_cu_code(i, j, cu_fld%data, p_fld%data, u_fld%data)
+        END DO 
+      END DO 
+      DO j=1,jstop+1
+        DO i=1,istop+1
+          CALL time_smooth_code(i, j, u_fld%data, unew_fld%data, uold_fld%data)
+        END DO 
+      END DO 
+    END SUBROUTINE invoke_loop1
+  END MODULE psy_test27_loop_swap"""
+
+    # Now swap the middle loops
+    swapped3, _ = swap.apply(swapped2.children[2])
+    psy.invokes.get('invoke_loop1').schedule = swapped3
+
+    assert str(psy.gen) == """  MODULE psy_test27_loop_swap
+    USE field_mod
+    USE kind_params_mod
+    IMPLICIT NONE
+    CONTAINS
+    SUBROUTINE invoke_loop1(cu_fld, p_fld, u_fld, unew_fld, uold_fld)
+      USE time_smooth_mod, ONLY: time_smooth_code
+      USE compute_cu_mod, ONLY: compute_cu_code
+      TYPE(r2d_field), intent(inout) :: cu_fld, p_fld, u_fld, unew_fld, uold_fld
+      INTEGER i
+      INTEGER j
+      INTEGER istop, jstop
+      !
+      ! Look-up loop bounds
+      istop = cu_fld%grid%simulation_domain%xstop
+      jstop = cu_fld%grid%simulation_domain%ystop
+      !
+      DO i=2,istop+1
+        DO j=2,jstop
+          CALL compute_cu_code(i, j, cu_fld%data, p_fld%data, u_fld%data)
+        END DO 
+      END DO 
+      DO i=2,istop+1
+        DO j=2,jstop
+          CALL compute_cu_code(i, j, cu_fld%data, p_fld%data, u_fld%data)
+        END DO 
+      END DO 
+      DO i=1,istop+1
+        DO j=1,jstop+1
+          CALL time_smooth_code(i, j, u_fld%data, unew_fld%data, uold_fld%data)
+        END DO 
+      END DO 
+    END SUBROUTINE invoke_loop1
+  END MODULE psy_test27_loop_swap"""
+
+    return
+
+
+def test_loop_swap_errors():
+    ''' Test loop swapping transform with incorrect parameters. '''
+
+    psy, invoke = get_invoke("test27_loop_swap.f90", 0)
+
+    schedule = invoke.schedule
+    swap = LoopSwapTrans()
+
+    # Test error if given node is not the outer loop of an at least
+    # double nested loop:
+    with pytest.raises(TransformationError):
+        swap.apply(schedule.children[0].children[0])
```

Comment:
Please add checks that it is the correct TransformationError that has been raised by doing e.g.:

    with pytest.raises(TransformationError) as excinfo:
        swap.apply(...)
    assert "node is not a loop" in str(excinfo)